src/App.tsx

```diff
@@ -1,61 +1,33 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './interfaces/todo';
+import { User } from './interfaces/user';
+import { Form } from './components/Form/Form';
 
 export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
+  const [todosArr, setTodosArr] = useState<Todo[]>(todosFromServer);
 
-          <span className="error">Please choose a user</span>
-        </div>
+  const addTodo = (todo: Todo) => setTodosArr([...todosArr, todo]);
 
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
+  const createTodosArr = (todos: Todo[], users: User[]) => {
+    const getUserById = (userId: number) => {
+      return users.find(user => user && user.id === userId);
+    };
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">
-            delectus aut autem
-          </h2>
+    return todos.map(todo => ({ ...todo, todoUser: getUserById(todo.userId) }));
+  };
 
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
+  const finalTodos = createTodosArr(todosArr, usersFromServer);
```

Comment:
better to make this function outside of component for initial todos state and after that change just state of todos.
in your implementation you make this operation during add each new todo

```
const getUserById = (userId: number) => {
  return usersFromServer.find(user => user && user.id === userId) || null;
};

const todosWithUser = todosFromServer
  .map(todo => ({ ...todo, todoUser: getUserById(todo.userId) }));
```